tests/unit/util/test.inc

```diff
@@ -231,6 +231,24 @@ static void test_CLAMP(void)
 		      0xffffffffaULL, "Unexpected clamp result");
 }
 
+static void test_IN_RANGE(void)
+{
+	zassert_true(IN_RANGE(0, 0, 0), "Unexpected IN_RANGE result");
+	zassert_true(IN_RANGE(1, 0, 1), "Unexpected IN_RANGE result");
+	zassert_true(IN_RANGE(1, 0, 2), "Unexpected IN_RANGE result");
+	zassert_true(IN_RANGE(-1, -2, 2), "Unexpected IN_RANGE result");
+	zassert_true(IN_RANGE(-3, -5, -1), "Unexpected IN_RANGE result");
+	zassert_true(IN_RANGE(0, 0, UINT64_MAX), "Unexpected clamp result");
+	zassert_true(IN_RANGE(UINT64_MAX, 0, UINT64_MAX), "Unexpected clamp result");
+	zassert_true(IN_RANGE(0, INT64_MIN, INT64_MAX), "Unexpected clamp result");
+	zassert_true(IN_RANGE(INT64_MIN, INT64_MIN, INT64_MAX), "Unexpected clamp result");
+	zassert_true(IN_RANGE(INT64_MAX, INT64_MIN, INT64_MAX), "Unexpected clamp result");
+
+	zassert_false(IN_RANGE(5, 0, 2), "Unexpected IN_RANGE result");
+	zassert_false(IN_RANGE(5, 10, 0), "Unexpected IN_RANGE result");
+	zassert_false(IN_RANGE(-1, 0, 1), "Unexpected IN_RANGE result");
```

Comment:
Should this also work if someone mixes floats with ints? What is expected behavior?